packages/contracts-bedrock/test/scripts/VerifyOPCM.t.sol

```diff
@@ -223,4 +237,65 @@ contract VerifyOPCM_Run_Test is VerifyOPCM_TestInit {
+        vm.setEnv("EXPECTED_UPGRADE_CONTROLLER", vm.toString(opcm.upgradeController()));
+
+        // Test that the immutable variables are correctly verified.
+        // Environment variables are set in setUp() to match the actual OPCM addresses.
+        bool result = harness.verifyOpcmImmutableVariables(opcm);
+        assertTrue(result, "OPCM immutable variables should be valid");
+    }
+
+    /// @notice Tests that the script fails when OPCM immutable variables are invalid.
+    /// We test this by setting expected addresses and mocking OPCM methods to return different addresses.
+    function test_verifyOpcmImmutableVariables_mismatch_fails() public {
+        // Coverage changes bytecode and causes failures, skip.
+        skipIfCoverage();
+
+        // Set expected addresses via environment variables
+        address expectedSuperchainConfig = address(0x1111);
+        address expectedProtocolVersions = address(0x2222);
+        address expectedSuperchainProxyAdmin = address(0x3333);
+        address expectedUpgradeController = address(0x4444);
+
+        vm.setEnv("EXPECTED_SUPERCHAIN_CONFIG", vm.toString(expectedSuperchainConfig));
+        vm.setEnv("EXPECTED_PROTOCOL_VERSIONS", vm.toString(expectedProtocolVersions));
+        vm.setEnv("EXPECTED_SUPERCHAIN_PROXY_ADMIN", vm.toString(expectedSuperchainProxyAdmin));
+        vm.setEnv("EXPECTED_UPGRADE_CONTROLLER", vm.toString(expectedUpgradeController));
+
+        // Mock all OPCM methods to return different addresses (causing mismatch)
+        vm.mockCall(
+            address(opcm), abi.encodeCall(IOPContractsManager.superchainConfig, ()), abi.encode(address(0x5555))
+        );
+        vm.mockCall(
+            address(opcm), abi.encodeCall(IOPContractsManager.protocolVersions, ()), abi.encode(address(0x6666))
+        );
+        vm.mockCall(
+            address(opcm), abi.encodeCall(IOPContractsManager.superchainProxyAdmin, ()), abi.encode(address(0x7777))
+        );
+        vm.mockCall(
+            address(opcm), abi.encodeCall(IOPContractsManager.upgradeController, ()), abi.encode(address(0x8888))
+        );
+
+        // Verify that immutable variables fail validation
+        bool result = harness.verifyOpcmImmutableVariables(opcm);
+        assertFalse(result, "OPCM with invalid immutable variables should fail verification");
+
+        // Clear mock calls and restore original environment variables to avoid test isolation issues
+        vm.clearMockedCalls();
+        // Reset environment variables to correct values (as set in setUp())
+        vm.setEnv("EXPECTED_SUPERCHAIN_CONFIG", vm.toString(address(opcm.superchainConfig())));
+        vm.setEnv("EXPECTED_PROTOCOL_VERSIONS", vm.toString(address(opcm.protocolVersions())));
+        vm.setEnv("EXPECTED_SUPERCHAIN_PROXY_ADMIN", vm.toString(address(opcm.superchainProxyAdmin())));
+        vm.setEnv("EXPECTED_UPGRADE_CONTROLLER", vm.toString(opcm.upgradeController()));
```

Comment:
It doesn't feel like this should need to be here, but I've confirmed that removing it causes a race condition with intermittent failures. Putting it into a reusable function as suggested above will make it feel cleaner.